initfiles/sbin/complete-uninstall.sh.in

```diff
@@ -100,21 +100,35 @@ if [ -n "${lpath}" -a -h "${lpath}" ] ; then
 fi
 
 if [ -e /etc/debian_version ]; then
-    echo "Removing DEB"
-    apt-get remove -y hpccsystems-platform
-    if [ $? -ne 0 ]; then
-       echo "An error has occured during removal.  Try \`apt-get -f remove hpccsystems-platform\` to fix this issue."
-       exit 1
+    pckname=$(apt-cache pkgnames hpccsystems-platform 2> /dev/null)
+    if [ -n "${pckname}" ] ; then
+        echo "Removing ${pckname}"
+        apt-get remove -y ${pckname}
+        if [ $? -ne 0 ]; then
+           echo "An error has occured during removal.  Try \`apt-get -f remove ${pckname}\` to fix this issue."
+           exit 1
+        fi
     fi
 elif [ -e /etc/redhat-release -o -e /etc/SuSE-release ]; then
-    echo "Removing RPM"
-    yum remove -y hpccsystems-platform
-    if [ $? -ne 0 ]; then
-       echo "An error has occured during removal.  Try to fix any issues and then run \`yum-complete-transaction\`."
-       exit 1
+    pckname=$(rpm -qa | grep hpccsystems-platform 2> /dev/null)
+    if [ -n "${pckname}" ] ; then
+        echo "Removing ${pckname}"
+        yum remove -y ${pckname}
+        if [ $? -ne 0 ]; then
+           echo "An error has occured during removal.  Try to fix any issues and then run \`yum-complete-transaction\`."
+           exit 1
+        fi
     fi
 fi
 
+if [ -z "${pckname}" ] ; then
```

Comment:
Lets add one more condition here -
```
f [ -z "${pckname}" -a $force -eq 0 ] ; then
```
That way you can run this script and if it does not find any pkg to remove you can still go on and remove the dirs and user.
What do you think ?